src/topology/category/Compactum.lean

```diff
@@ -0,0 +1,433 @@
+    specialize cond Aᶜ a hA,
+    rw ultrafilter_iff_compl_mem_iff_not_mem.mp F.2 at cond,
+    contradiction },
+  -- If A ∈ F, then x ∈ cl A.
+  have claim2 : ∀ (A : set X), A ∈ F.1 → x ∈ cl A,
+  { intros A hA,
+    exact claim1 (cl A) (is_closed_cl A) (F.val.sets_of_superset hA (subset_cl A)) },
+  -- T0 is closed under intersections.
+  have claim3 : ∀ (S1 S2 ∈ T0), S1 ∩ S2 ∈ T0,
+  { rintros S1 S2 ⟨S1, hS1, rfl⟩ ⟨S2, hS2, rfl⟩,
+    exact ⟨S1 ∩ S2, F.1.inter_sets hS1 hS2, by simp [basic_inter]⟩ },
+  -- For every S ∈ T0, the intersection AA ∩ S is nonempty.
+  have claim4 : ∀ (S ∈ T0), (AA ∩ S).nonempty,
+  { rintros S ⟨S, hS, rfl⟩,
+    rcases claim2 _ hS with ⟨G, hG, hG2⟩,
+    exact ⟨G, hG2, hG⟩ },
+  -- Every element of T0 is nonempty.
+  have claim5 : ∀ (S ∈ T0), set.nonempty S,
+  { rintros S ⟨S, hS, rfl⟩,
+    exact ⟨F, hS⟩ },
+  -- Every element of T2 is nonempty.
+  have claim6 : ∀ (S ∈ T2), set.nonempty S,
+  { suffices : ∀ S ∈ T2, S ∈ T0 ∨ ∃ Q ∈ T0, S = AA ∩ Q,
+    { intros S hS,
+      specialize this _ hS,
+      cases this,
+      { exact claim5 S this },
+      { rcases this with ⟨Q, hQ, rfl⟩,
+        exact claim4 Q hQ}},
+    intros S hS,
+    apply finite_inter_closure_insert,
+    { split,
+      { use set.univ,
+        refine ⟨filter.univ_sets _, _⟩,
+        ext,
+        refine ⟨_, by tauto⟩,
+        { intro, apply filter.univ_sets, } },
+      { exact claim3} },
+    { exact hS} },
+  -- It suffices to show that the intersection of any finite subset of T1 is nonempty.
+  suffices : ∀ (F : fsu), ↑F ⊆ T1 → (⋂₀ ι F).nonempty,
+  { obtain ⟨G,h1,hG⟩ := exists_ultrafilter_of_finite_inter_nonempty _ this,
+    let GG : ultrafilter (ultrafilter X) := ⟨G, hG⟩,
+    have c1 : X.join GG = F,
+    { suffices : (X.join GG).1 ≤ F.1,
+      { ext1,
+        exact is_ultrafilter.unique F.2 (join X GG).2.1 this},
+      intros P hP,
+      exact h1 (or.inr ⟨P, hP, rfl⟩) },
+    have c2 : ultrafilter.map X.str GG = X.incl x,
+    { suffices : (ultrafilter.map X.str GG).1 ≤ (X.incl x).1,
+      { ext1,
+        exact is_ultrafilter.unique (incl X x).2 (ultrafilter.map (str X) GG).2.1 this },
+      intros P hP,
+      apply filter.sets_of_superset _ (h1 (or.inl rfl)),
+      rintros x ⟨rfl⟩,
+      exact hP },
+    simp [←c1,c2] },
+  -- Finish...
+  intros T hT,
+  refine claim6 _ (finite_inter_mem (finite_inter_closure_has_finite_inter _) _ _),
+  intros t ht,
+  exact finite_inter_closure.basic (@hT t ht)
+end
+
+lemma le_nhds_of_str_eq {X : Compactum} (F : ultrafilter X) (x : X) :
+  X.str F = x → F.1 ≤ nhds x := λ h, le_nhds_iff.mpr (λ s hx hs, hs _ $ by rwa h)
+
+-- All the hard work above boils down to this t2_instance.
+instance {X : Compactum} : t2_space X :=
+begin
+  rw t2_iff_ultrafilter,
+  intros _ _ _ hF hx hy,
+  replace hx := str_eq_of_le_nhds ⟨_,hF⟩ _ hx,
+  replace hy := str_eq_of_le_nhds ⟨_,hF⟩ _ hy,
+  cc,
+end
+
+-- Any continuous morphism of compacta is a morphism of algebras for the ultrafilter monad.
```

Comment:
I think this comment is the opposite of what's proved here? The statement given says that a morphism of algebras gives a continuous function on the spaces.

Reply:
Of course, you're right. I was confusing this with `continuous_to_hom` below.

Reply:
Fixed.